File: src/mod10_design_pattern/lesson7_adapter/java/Main.java
package mod10_design_pattern.lesson7_adapter.java;

public class Main {
    public static void main(String[] args) {

        Spell harry = new Wizard();
        harry.cast();

        Muggle dudley = new Muggle();
        MuggleAdapter adapter = new MuggleAdapter(dudley);
        adapter.cast();


    }
}

File: src/mod10_design_pattern/lesson7_adapter/kotlin/Main.kt
package mod10_design_pattern.lesson7_adapter.kotlin

fun main() {

    val harry = Wizard()
    harry.cast()

    val dudley = Muggle()
    val adapter=MuggleAdapter(dudley)
    dudley.useMuggleSkills()

    val spell: Spell = harry
    val spell2: Spell = adapter

    adapter.cast()

}